src/llm_utils.py

```diff
@@ -0,0 +1,190 @@
+    
+    if current_block:
+        comment_blocks.append('\n'.join(current_block))
+
+    for block in comment_blocks:
+        comment_data = {
+            'path': None,
+            'line': None,
+            'body': []
+        }
+        
+        current_section = None
+        
+        for line in block.split('\n'):
+            line = line.rstrip()
+            
+            if line.startswith('FILE:'):
+                comment_data['path'] = line.split('FILE:', 1)[1].strip()
+            elif line.startswith('LINE:'):
+                try:
+                    comment_data['line'] = int(line.split('LINE:', 1)[1].strip())
+                except (ValueError, IndexError):
+                    continue
+            elif line.startswith('COMMENT:'):
+                comment_data['body'].append(line.split('COMMENT:', 1)[1].strip())
+            elif line.startswith('SUGGESTION:'):
+                suggestion = line.split('SUGGESTION:', 1)[1].strip()
+                comment_data['body'].append(f"\n```suggestion\n{suggestion}\n```")
+            elif line and any([line.startswith(s) for s in ['FILE:', 'LINE:', 'COMMENT:', 'SUGGESTION:']]):
+                continue  # Skip empty lines after headers
+            else:
+                if comment_data['body']:
+                    comment_data['body'].append(line)
+
+        # Validate and format
+        if comment_data['path'] and comment_data['line'] is not None and comment_data['body']:
+            try:
+                formatted_body = '\n'.join(comment_data['body'])
+                # Remove empty suggestion blocks
+                formatted_body = formatted_body.replace('```suggestion\n\n```', '')
+                review_comments.append({
+                    'path': comment_data['path'],
+                    'line': comment_data['line'],
+                    'body': formatted_body
+                })
+            except KeyError:
+                continue
+
+    return review_comments
+
+
+# def parse_llm_response(response: str) -> List[Dict]:
+#     """
+#     Parse the LLM response and format it into review comments
+#     Returns a list of structured comment objects
+#     """
+#     review_comments = []
+    
+#     # Split response into individual comments
+#     comment_blocks = response.split('---')
+    
+#     for block in comment_blocks:
+#         lines = [line.strip() for line in block.split('\n') if line.strip()]
+#         if not lines:
+#             continue
+            
+#         comment_data = {
+#             'body': '',
+#             'path': None,
+#             'position': 1,  # Default to first line if not specified
+#             'suggestion': None
+#         }
+        
+#         current_section = None
+        
+#         for line in lines:
+#             if line.startswith('FILE:'):
+#                 comment_data['path'] = line.split('FILE:')[1].strip()
+#             elif line.startswith('LINE:'):
+#                 try:
```

Comment:
The comment "Skip empty lines after headers" can be misleading, because it is actually skipping the current line if it starts with 'FILE:', 'LINE:', 'COMMENT:', or 'SUGGESTION:'. This prevents extracting new information.

```suggestion
Remove this line
```
```
            elif line and any([line.startswith(s) for s in ['FILE:', 'LINE:', 'COMMENT:', 'SUGGESTION:']]):
                continue  # Skip empty lines after headers
```